### PR TITLE
Make pyproject.toml and poetry.lock consistent.

### PR DESCRIPTION
**Pull request by netsettler**
* pyproject.toml and poetry.lock are out of sync on master for foursight-core, so did a `poetry update dcicutils` to get a new `poetry.lock`. This pulled in `dcicutils 4.4.0`, so I updated `pyproject.toml` to require that version, since it's harmless and it speeds locking.
* Made some PEP8 updates so that `make lint` would pass.  Noticed some issues that might want to be repaired later and made comments about them, but I think the fixes should not go in this PR, since the main purpose of this PR is to get `poetry.lock` consistent on master and that should be done soon.
* Updated CHANGELOG and bumped version.

### CODE REVIEW

File: foursight_core/deploy.py [hunk 1]
```diff
@@ -64,7 +64,12 @@ def build_config(cls, stage, identity=None, stack_name=None,
         """ Builds the chalice config json file. See: https://aws.github.io/chalice/topics/configfile"""
         # dmichaels/2022-07-22/C4-826:
         # Removed value from the Foursight CloudFormation template; get from GAC/etc at runtime.
-        ignored(stage)
+        # TODO: Is it in fact correct that we are not using rds_name here? The changelog for 1.0.0 doesn't mention it,
+        #       and should be retroactively modified to explain any changes in environment variable use due to the
+        #       1.0.0 change so that users can better adapt to this upgrade. (I noticed that one or more FF
+        #       environments were missing this env variable and was searching all over for clues about why
+        #       that might be.) -kmp 19-Aug-2022
```
Comment (dmichaels-harvard) [inline foursight_core/deploy.py:71]: Oh, yes, sorry about that; RDS_NAME will now be in the GAC (with newer, in PR, 4dn-cloud-infra); I will add that the the CHANGELOG.rst.

File: foursight_core/deploy.py [hunk 2]
```diff
@@ -173,6 +178,9 @@ def build_config_and_package(cls, args, identity=None, stack_name=None, trial_cr
             environment variables, a list of security group ids, and a list of subnet ids. Finally, packages as a
             Cloudformation template."""
 
+        # TODO: This passes rds_name through, but build_config is no longer using it. Is that correct?
+        #       Should this be just marking that arg ignored or just not receiving such an arg?
+        #       Something is weird in that data flow chain. -kmp 19-Aug-2022
```
Comment (dmichaels-harvard) [inline foursight_core/deploy.py:183]: Oh, yes, sorry about that; RDS_NAME will now be in the GAC (with newer, in PR, 4dn-cloud-infra); I will add that the the CHANGELOG.rst. I should remove the rds_name arg now; will take care of that.

**Review by dmichaels-harvard — APPROVED** [review]

**Comment by netsettler** [comment]
@dmichaels , at this point I'm not sure this PR serves a lot of purpose, but if I close it I'm worried about dropping the thing about the `RDS_NAME` dataflow. Is ignoring that `rds_name` arg right, or should it be passed through somehow. Other than that, I don't think there's any utility here and I should just close this. Do you agree?

...
From @dmichaels: The `rds_name` is no longer needed. I will remove it from foursight-core/deploy.py (`build_config` and `build_config_and_package`).

**Comment by netsettler** [comment]
I don't think there's any useful content left in this PR, so I'm closing without a merge.

**Pull request closed without merging**